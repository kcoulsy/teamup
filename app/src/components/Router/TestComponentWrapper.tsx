import React from 'react';

const TestComponentWrapper: React.FunctionComponent = ({ children }) => {
  return <div>{children}</div>;
};

export default TestComponentWrapper;
