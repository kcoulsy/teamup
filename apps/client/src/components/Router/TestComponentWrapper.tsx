import React, { PropsWithChildren } from 'react';

function TestComponentWrapper({ children }: PropsWithChildren<{}>) {
  return <div>{children}</div>;
}

export default TestComponentWrapper;
