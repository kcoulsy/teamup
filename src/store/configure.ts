import { configureStore } from '@reduxjs/toolkit';

// Import Reducers
import authReducer from '../reducers/auth';
import teamReducer from '../reducers/team';

const store = configureStore({
    reducer: {
        // reducers here
        auth: authReducer,
        team: teamReducer,
    },
});

export type RootState = ReturnType<typeof store.getState>;

export default store;
