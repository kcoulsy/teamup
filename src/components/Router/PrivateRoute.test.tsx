import React from 'react';
import { shallow } from 'enzyme';
import {
  privateRouteRender,
  mapStateToProps,
  PrivateRoute,
} from './PrivateRoute';
import { shallowToJson } from 'enzyme-to-json';
import isLoggedIn from '../../helpers/isLoggedIn';
import { RootState } from '../../store/configure';
import * as H from 'history';
import TestComponentWrapper from './TestComponentWrapper';

vi.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: vi.fn(),
  }),
}));

describe('render PrivateRoute component', () => {
  const child = <p>JSX Child</p>;
  const location: H.Location<any> = {
    pathname: '',
    search: '',
    state: '',
    hash: '',
    key: '',
  };

  it('should render PrivateRoute component', () => {
    const wrapper = shallow(<PrivateRoute isLoggedIn={false} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle PrivateRouteRender when logged in', () => {
    const isLoggedIn = true;
    const wrapper = shallow(
      <TestComponentWrapper>
        {privateRouteRender(isLoggedIn, child)({ location })}
      </TestComponentWrapper>
    );

    expect(wrapper.contains(child)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle PrivateRouteRender when not logged in', () => {
    const isLoggedIn = false;
    const wrapper = shallow(
      <TestComponentWrapper>
        {privateRouteRender(isLoggedIn, child)({ location })}
      </TestComponentWrapper>
    );

    expect(wrapper.contains(child)).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return correct props from store', () => {
    const mockState: RootState = {
      auth: {
        attemptingLogin: false,
        token: 'testtoken',
        loginAttemptFailed: false,
        appInitialising: false,
        appInitialised: false,
        attemptingRegister: false,
      },
    };

    expect(mapStateToProps(mockState)).toEqual({
      isLoggedIn: isLoggedIn(mockState.auth.token),
    });
  });
});
