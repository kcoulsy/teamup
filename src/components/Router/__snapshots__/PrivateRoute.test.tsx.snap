// Vitest Snapshot v1

exports[`render PrivateRoute component > should handle PrivateRouteRender when logged in 1`] = `ShallowWrapper {}`;

exports[`render PrivateRoute component > should handle PrivateRouteRender when not logged in 1`] = `ShallowWrapper {}`;

exports[`render PrivateRoute component > should render PrivateRoute component 1`] = `
<Component
  render={[Function]}
/>
`;

exports[`render PrivateRoute component should handle PrivateRouteRender when logged in 1`] = `
<div>
  <p>
    JSX Child
  </p>
</div>
`;

exports[`render PrivateRoute component should handle PrivateRouteRender when not logged in 1`] = `
<div>
  <Component
    to={
      Object {
        "pathname": "/login",
        "state": Object {
          "from": Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
            "state": "",
          },
        },
      }
    }
  />
</div>
`;

exports[`render PrivateRoute component should render PrivateRoute component 1`] = `
<Component
  render={[Function]}
/>
`;
